helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -412,6 +429,30 @@ private void refreshBaseline(ResourceControllerDataProvider clusterData,
     return newAssignment;
   }
 
+  private void measureBaselineDivergenceGauge(Map<String, ResourceAssignment> baseline,
+      Map<String, ResourceAssignment> bestPossibleAssignment) {
+    MapDifference<String, ResourceAssignment> mapDifference =
+        Maps.difference(baseline, bestPossibleAssignment);
+    Map<String, ResourceAssignment> matchedAssignment = mapDifference.entriesInCommon();
+    Map<String, MapDifference.ValueDifference<ResourceAssignment>> differentAssignment =
+        mapDifference.entriesDiffering();
```

Comment:
This diff only shows if anything diff in the ResourceAssignment, right?
I think we need to check at the single partition level, at least.

Reply:
Right. Resolved.